lib/utils/VCUtils.ts

```diff
@@ -34,27 +31,32 @@ export function definitionVersionDiscovery(presentationDefinition: IPresentation
   JsonPathUtils.changePropertyNameRecursively(presentationDefinitionCopy, '_const', 'const');
   JsonPathUtils.changePropertyNameRecursively(presentationDefinitionCopy, '_enum', 'enum');
   const data = { presentation_definition: presentationDefinitionCopy };
-  let result = validatePDv2(data);
-
-  if (result) {
+  if (validatePDv2(data)) {
     return { version: PEVersion.v2 };
   }
-  // Errors are added to the validation method, but not typed correctly
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  const v2Errors = validatePDv2.errors;
+  const v2Errors = validatePDv2.errors ?? undefined;
 
-  result = validatePDv1(data);
-  if (result) {
+  if (validatePDv1(data)) {
     return { version: PEVersion.v1 };
   }
+  const v1Errors = validatePDv1.errors ?? undefined;
 
-  // Errors are added to the validation method, but not typed correctly
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  const v1Errors = validatePDv1.errors;
+  return {
+    error: 'This is not a valid PresentationDefinition',
+    v1Errors,
+    v2Errors,
+  };
+}
 
-  return { error: 'This is not a valid PresentationDefinition', v1Errors, v2Errors };
+export function formatValidationError(error: ValidationError): string {
+  return `${error.instancePath || '/'}: ${error.message}${error.params.additionalProperty ? ` (${error.params.additionalProperty})` : ''}`;
+}
+
+export function formatValidationErrors(errors: ValidationError[] | undefined): string | undefined {
+  if (!errors?.length) {
+    return undefined;
+  }
+  return errors.map(formatValidationError).join('\n  ');
```

Comment:
Is this correct? Joining with linebreak + empty space?

Reply:
Yes the error lines offset a bit under the main error, like a tab.